### PR TITLE
ci: Validate load options methods in nodes-base

### PR DESCRIPTION
**Pull request by ivov**
During CI, generate UI types with `loadOptionsMethods` and validate that all referenced methods have been defined and that all defined methods are being used. ESLint cannot access the AST of other files while running on a given file and keep info in memory throughout the run and also guarantee that required files are parsed in order, so a CI script reusing Adi's loader logic looks like a better fit.

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
Considering that `n8n-generate-ui-types-with-methods` isn't noticeably slower than `n8n-generate-ui-types`, why not combine the two files, and always generate the additional json files?

**Comment by ivov** [comment]
> why not combine the two files, and always generate the additional json files?

Aimed to have zero impact on the regular build and avoid adding complexity to your script, at the cost of a bit of duplication, but I was on the fence on this. I'll combine them if you prefer.

File: packages/nodes-base/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
     "typecheck": "tsc",
     "build": "tsc -p tsconfig.build.json && gulp build:icons && gulp build:translations && pnpm build:metadata",
     "build:translations": "gulp build:translations",
-    "build:metadata": "pnpm n8n-generate-known && pnpm n8n-generate-ui-types",
+    "build:metadata": "pnpm n8n-generate-known && pnpm n8n-generate-ui-types && node ./scripts/validate-load-options-methods.js",
```
Comment (netroy) [inline packages/nodes-base/package.json:22]: can we please move this script to the `lint` task instead

**Review by netroy — APPROVED** [review]

**Comment by netroy** [comment]
Should we address these ?
image

**Comment by ivov** [comment]
> Should we address these ?

I set these as warnings because it's currently dead code but otherwise harmless. I'll leave it to the nodes team to decide.

**Pull request merged**